app/src/marketplace/components/Modal/ChooseAccessPeriodDialog/index.jsx

```diff
@@ -7,255 +7,222 @@ import cx from 'classnames'
-            setPriceInDai(BN(daiValue).isNaN() ? 'N/A' : formatDecimals(daiValue, paymentCurrencies.DAI).toString())
+    }, [paymentCurrency, priceInUsd, currentPrice])
+
+    const setExternalPrices = useDebounced(useCallback(async ({
+        dataPerUsd: perUsd,
+        priceInData: inData,
+        priceInUsd: inUsd,
+        paymentCurrency: currency,
+    }) => {
+        setLoading(true)
 
-            setLoading(false)
+        let price
+        let usdEstimate
+        if (currency === paymentCurrencies.ETH) {
+            price = await uniswapDATAtoETH(inData.toString(), true)
+            usdEstimate = await uniswapETHtoDATA(price.toString(), true)
+            usdEstimate = usdEstimate.dividedBy(Number(perUsd) || 1)
+        } else if (currency === paymentCurrencies.DAI) {
+            price = await uniswapDATAtoDAI(inData.toString(), true)
+            usdEstimate = price
         } else {
-            setPriceInEth('-')
-            setPriceInDai('-')
+            price = inData
+            usdEstimate = inUsd
         }
-    }, [dataPerUsd, isValidTime, priceInData])
 
-    const getAccountBalance = useCallback(async () => {
-        setLoading(true)
-        const [eth, data, dai] = await getBalances()
-        setLoading(false)
+        setCurrentPrice(price)
+        setApproxUsd(usdEstimate)
 
-        switch (paymentCurrency) {
-            case paymentCurrencies.ETH:
-                return setEthBalance(formatDecimals(eth, paymentCurrencies.ETH))
-            case paymentCurrencies.DATA:
-                return setDataBalance(formatDecimals(data, paymentCurrencies.DATA))
-            case paymentCurrencies.DAI:
-                return setDaiBalance(formatDecimals(dai, paymentCurrencies.DAI))
-            default:
-                return undefined
-        }
-    }, [paymentCurrency])
+        setLoading(false)
+    }, []), 250)
 
-    const approxUsd = () => {
-        if (paymentCurrency === paymentCurrencies.ETH) {
-            return ethPriceInUsd
-        } else if (paymentCurrency === paymentCurrencies.DAI) {
-            return priceInDai
-        }
-        return priceInUsd
-    }
+    const displayPrice = useMemo(() => (
+        BN(currentPrice).isNaN() ? 'N/A' : formatDecimals(currentPrice, paymentCurrency)
+    ), [currentPrice, paymentCurrency])
 
-    useEffect(() => {
-        getProductPrices()
-    }, [getProductPrices])
+    const displayApproxUsd = useMemo(() => (
+        BN(approxUsd).isNaN() ? 'N/A' : formatDecimals(approxUsd, contractCurrencies.USD)
+    ), [approxUsd])
 
     useEffect(() => {
-        getAccountBalance()
-    }, [getAccountBalance, account])
+        setExternalPrices({
+            dataPerUsd,
+            priceInData,
+            priceInUsd,
+            paymentCurrency,
+        })
+    }, [setExternalPrices, dataPerUsd, priceInData, paymentCurrency, priceInUsd])
```

Comment:
the dependencies are slightly clearer now, updates everytime either `dataPerUsd`, `priceInData`, `paymentCurrency` or `priceInUsd` updates

Reply:
much better 👍